src/routes/root/Transactions.tsx

```diff
@@ -1,9 +1,127 @@
+import { useEffect, useState } from 'react';
+import { Contract } from 'web3';
+import { EventLog } from 'web3-eth-contract';
 import { Heading } from '../../components/Heading';
+import { useStorage } from '../../context/StorageContext';
+import { useEnsuredWallet } from '../../context/WalletContext';
+import { bridgeABI } from '../../contracts';
+import { getBridgeAddress, getDomainID, NetworksChainID } from '../../utils/network';
+
+interface DepositLogReturnValues {
+  data: string;
+  depositNonce: bigint;
+  destinationDomainID: bigint;
+  handlerResponse: string;
+  resourceID: string;
+  user: string;
+}
+
+enum TX_STEPS {
+  Initializig,
+  SendingFunds,
+  Success,
+}
 
 export function Transactions(): JSX.Element {
+  const wallet = useEnsuredWallet();
+  const storage = useStorage();
+
+  const [steps, setSteps] = useState<TX_STEPS>(TX_STEPS.Initializig);
+  const [logCompareData, setLogCompareData] = useState<{
+    originDomainID: bigint;
+    depositNonce: bigint;
+  } | null>(null);
+
+  useEffect(() => {
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    (async () => {
+      if (steps === TX_STEPS.Initializig) {
+        const accounts = await wallet.web3.eth.getAccounts();
+        const networkChainId = await wallet.web3.eth.getChainId();
+
+        // @ts-ignore
+        const depositAdapterContract: Contract<typeof bridgeABI> =
+          new wallet.web3.eth.Contract(
+            bridgeABI,
+            getBridgeAddress(wallet.web3.utils.toHex(networkChainId) as NetworksChainID),
+            {
+              from: accounts[0],
+              provider: wallet.web3.provider,
+            },
+          );
+        const transactionHash = storage.data.txReceiptHash;
+
+        const logs = await depositAdapterContract.getPastEvents('Deposit');
```

Comment:
you should pass option to specify block range, but you probably don't need this if you are gonna listen to that event immediately after tx is submitted